CONTRIBUTING.md

```diff
@@ -30,6 +34,27 @@ git commit -a
 git push origin my-new-feature
 ```
 
+### Code Review
+
+All contributions to Precaution must be reviewed and approved by a maintainer (distinct from the author of the pull request). Pull Requests will be merged to the master branch once one or more maintainers have Approved the Pull Request and the _ready-to-merge_ label has been added.
+
+This final step of adding the _ready-to-merge_ label allows for a range of final check scenarios once the code is deemed ready by a reviewer. Examples of final checks that may occur are:
+- when a Reviewer wants another project maintainer to approve the change
+- when the Submitter needs to squash, rebase, or otherwise clean up their changes before the merge
+
+If a Pull Request is Approved without the _ready-to-merge_ label being added the reviewer must add a comment indicating what must happen before the change can be merged so that the Submitter knows what is gating acceptance of their contribution.
+
+The expected code review process is:
+
+- Submitter creates Pull Request
+- (_optional_) Reviewer provides feedback
+- (_optional_) Submitter addresses feedback, adding further patches to the pull request
+- Reviewer Approves merging of the changes in the Pull Request
+- (_optional_) Final confirmation (from a second reviewer, following rebase, etc)
+- Reviewer adds _ready-to-merge_ label
+
+__Note__: the _'work-in-progress'_ label is used to indicate that a Pull Request is not ready for merging, but that feedback and discussion on the Pull Request's contents are desirable.
```

Comment:
Suggest Draft PRs used instead which are better since regular contributors can't add labels to their PRs.

Reply:
Are you suggesting something like:

> __Note__: the _work-in-progress_ label is used to indicate draft Pull Requests from the maintainers which aren't yet ready for merging.

?

Reply:
When a PR is created through GitHub, you have the option to create as a draft.  

The work-in-progress label can only be marked by the project maintainer, so not very useful for a regular contributor.  I'd suggest they use drafts or prefix the commit title with `WIP:`

Reply:
https://github.blog/2019-02-14-introducing-draft-pull-requests/

Reply:
Now that draft PR's are a feature of the platform it doesn't really make sense to include this note in the documentation, I'll remove it.